api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -511,6 +511,20 @@ default NDArray get(NDIndex index) {
         return getNDArrayInternal().getIndexer().get(this, index);
     }
 
+    /**
+     * Returns a partial {@code NDArray}.
+     *
+     * @param index the boolean or int {@code NDArray} that indicates what to get
+     * @return the partial {@code NDArray}
+     */
+    default NDArray get(NDArray index) {
+        if (index.getDataType() == DataType.BOOLEAN) {
+            return get(new NDIndex().addBooleanIndex(index));
+        } else {
+            return take(index);
```

Comment:
Probably in the next PR, change this to `return get(new NDIndex().addTakeIndex(index))` or combine both branches into a single path `return get(new NDIndex().addArrayIndex(index))` and handle it on the NDIndexing side.